### PR TITLE
Master subscription perf yti

### PR DESCRIPTION
**Pull request by tivisse**
Com PR: https://github.com/odoo/odoo/pull/88621
Ent PR: https://github.com/odoo/enterprise/pull/26153

### CODE REVIEW

File: addons/rating/models/rating_mixin.py [hunk 1]
```diff
@@ -27,9 +27,24 @@ class RatingMixin(models.AbstractModel):
 
     @api.depends('rating_ids.rating', 'rating_ids.consumed')
     def _compute_rating_last_value(self):
+        # Pure SQL instead of calling read_group to allow ordering array_agg
+        if not self.ids:
+            self.rating_last_value = 0
+            return
+        self.env.cr.execute("""
+            SELECT
+                array_agg(rating ORDER BY write_date DESC) AS "ratings",
```
Comment (Williambraecky) [inline addons/rating/models/rating_mixin.py:36]: Is this faster than a subselect? I wonder what is the cost of `array_agg` vs an indexed scan.
PS: a quick explain analyze on both seems to indicate that it's actually quite similar (without data though).

```suggestion
                (SELECT rating FROM rating_rating r2 WHERE r2.res_id = r.res_id AND r2.res_model = %s AND consumed = true ORDER BY write_date DESC LIMIT 1) AS "ratings",
```

File: addons/sale/models/sale_order.py [hunk 2]
```diff
@@ -768,8 +769,11 @@ def action_confirm(self):
                 'It is not allowed to confirm an order in the following states: %s'
             ) % (', '.join(self._get_forbidden_state_confirm())))
 
+        mapped_subscriptions = defaultdict(lambda: self.env['sale.order'])
         for order in self.filtered(lambda order: order.partner_id not in order.message_partner_ids):
-            order.message_subscribe([order.partner_id.id])
+            mapped_subscriptions[order.partner_id.id] += order
```
Comment (Williambraecky) [inline addons/sale/models/sale_order.py:774]: We may as well remove the `for.. in x.filtered` here:
```suggestion
            if order.partner_id in order.message_partner_ids:
                continue
            mapped_subscriptions[order.partner_id.id] += order
```

File: addons/sale/models/sale_order.py [hunk 3]
```diff
@@ -768,8 +769,11 @@ def action_confirm(self):
...
+        for partner_id, orders in mapped_subscriptions.items():
+            orders.message_subscribe([partner_id])
```
Comment (Williambraecky) [inline addons/sale/models/sale_order.py:776]: We can store records directly and use `.ids`

```suggestion
            mapped_subscriptions[order.partner_id] += order
        for partner_id, orders in mapped_subscriptions.items():
            orders.message_subscribe(partner_id.ids)
```

File: addons/rating/models/rating_mixin.py (hunk 1 above)
Comment (Feyensv) [reply addons/rating/models/rating_mixin.py:36]: Isn't this ordering perf problem fixed by the private `_read_group` added in #84908 ?
(Not 100% sure, but I thought of this change when reading your above comment and diff)

**Comment by C3POdoo** [comment]
Dear @tivisse,

Thank you for your contribution but we are closing it due to inactivity.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**